src/Features/CSharp/Portable/ConvertToRawString/ConvertRegularStringToRawStringCodeRefactoringProvider.cs

```diff
@@ -247,19 +260,110 @@ private static bool HasLeadingWhitespace(VirtualCharSequence characters)
+                // multi-line strings that don't want any indentation and have the start/end of the string at the same
+                // level (like unit tests).
+                //
+                // In this case, figure out what indentation we're normally like to put this string.  Update *both* the
+                // contents *and* the starting quotes of the raw string.
+                var indenter = parsedDocument.LanguageServices.GetRequiredService<IIndentationService>();
+                var indentationVal = indenter.GetIndentation(parsedDocument, tokenLine.LineNumber, indentationOptions, cancellationToken);
+
+                var indentation = indentationVal.GetIndentationString(parsedDocument.Text, indentationOptions);
+                return ConvertToMultiLineRawIndentedString(indentation, addIndentationToStart: true);
+            }
+            else
+            {
+                // otherwise this was a string literal on a line that already contains contents.  Or it's a string
+                // literal on its own line, but indented some amount.  Figure out the indentation of the contents from
+                // this, but leave the string literal starting at whatever position it's at.
+                var indentation = token.GetPreferredIndentation(parsedDocument, indentationOptions, cancellationToken);
+                return ConvertToMultiLineRawIndentedString(indentation, addIndentationToStart: false);
             }
 
-            var indentationOptions = new IndentationOptions(formattingOptions);
-            var parsedDocument = await ParsedDocument.CreateAsync(document, cancellationToken).ConfigureAwait(false);
-            var indentation = token.GetPreferredIndentation(parsedDocument, indentationOptions, cancellationToken);
-            return ConvertToMultiLineRawIndentedString(indentation, token, formattingOptions, characters);
+            SyntaxToken ConvertToSingleLineRawString()
+            {
+                // Have to make sure we have a delimiter longer than any quote sequence in the string.
+                var longestQuoteSequence = GetLongestQuoteSequence(characters);
+                var quoteDelimiterCount = Math.Max(3, longestQuoteSequence + 1);
+
+                using var _ = PooledStringBuilder.GetInstance(out var builder);
+
+                builder.Append('"', quoteDelimiterCount);
+
+                foreach (var ch in characters)
+                    ch.AppendTo(builder);
+
+                builder.Append('"', quoteDelimiterCount);
+
+                return SyntaxFactory.Token(
+                    token.LeadingTrivia,
+                    SyntaxKind.SingleLineRawStringLiteralToken,
+                    builder.ToString(),
+                    characters.CreateString(),
+                    token.TrailingTrivia);
+            }
+
+            SyntaxToken ConvertToMultiLineRawIndentedString(
+                string indentation,
+                bool addIndentationToStart)
```

Comment:
this helper was inlined.  though the addIndentationToStart part is new.  it represents the new behavior we want.